hooks/96-auto-manage-uefi-entries.install

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
+check_if_uefi_entry_exists () {
+	for entry in $(efibootmgr); do
+
+		if [[ "$entry" == *"($efi_path"* ]] && [[ "$entry" == *"$partition_uuid"* ]]; then
+			
+			return 1
+		fi
+			
+	done
+	return 0
+}
+
+add_uefi_entry () (
+	for efi in $partition_efis; do
+		
+		# Prepare path which will be inserted into efibootmgr
+		efi_path=$(echo "${efi//$partition_mount}" | sed 's/\//\\/g')
+
+		# Add entry if it doesn't exist
+		if check_if_uefi_entry_exists; then
+
+			echo "Adding UEFI entry for $efi_path found on $partition..."
+			efibootmgr --create --disk /dev/$partition --label "Gentoo $partition" --loader "$efi_path" -u $kenrel_commands &>/dev/null
```

Comment:
Since this is going to run for all `.efi` files we should then add some naming logic. Unified Kernel Images are always named $ENTRY_TOKEN-$KERNEL_VERSION.efi. Everything in the `EFI/Linux` directory we can therefore give the `--label $ENTRY_TOKEN-$KERNEL_VERSION`.  But there may also be efi files belonging to boot loaders, Windows, or other things on this ESP, either we have to ignore those (and only consider files in locations where `installkernel` may install them: `EFI/Linux` (uki layout) and `ENTYRY_TOKEN` (bls layout)) , or we have to add some logic to give these entries proper names.

My preference would be to only manage entries that we know are ours, i.e. only entries that match our `ENTRY_TOKEN`. Otherwise we add a lot of complexity.

As a side note, there is a type here `$kenrel_commands` -> `$kernel_commands`

Reply:
Thx, for highlighting the typo.

I completely agree with you on this in all points.

As for how to distinguish them, I would stand on the side of acknowledging only kernels matching our naming scheme.

To be fair I don't want to limit it to only certain directories. I like the bigger flexibility of it. But I will not die on this hill.

Reply:
> As for how to distinguish them, I would stand on the side of acknowledging only kernels matching our naming scheme.
> 
> To be fair I don't want to limit it to only certain directories. I like the bigger flexibility of it. But I will not die on this hill.

An overview of the currently possible kernel names and locations is here: https://wiki.gentoo.org/wiki/Installkernel#USE_configuration_to_boot_layout_mapping

I'm not fundamentally opposed to scanning all directories on the ESP, but it does mean that we have to add some logic that sets a sensible `--label` for those entries, e.g. for `EFI/systemd/systemd-bootx64.efi, we should then set something like "systemd-boot".

Reply:
> sets a sensible `--label` for those entries

...containing `Gentoo`